viewing_party/party.py

```diff
@@ -1,23 +1,163 @@
+    if not title_check or not genre_check or not rating_check:
+        return None
+    movie = {}
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+    return movie
+    
+
+def add_to_watched(user_data, movie):
+    user_data_watched_list = user_data["watched"]
+    user_data_watched_list.append(movie)
+    return user_data
+    
+
+def add_to_watchlist(user_data, movie):
+    user_data_watchlist = user_data["watchlist"]
+    user_data_watchlist.append(movie)
+    return user_data
 
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    lengh_list = len(user_data["watched"])
+    if lengh_list < 1:
+        return 0.0
+    rating_added = 0
+    for element in user_data["watched"]:
+        rating_added += element["rating"]
+    average_rating = rating_added/lengh_list
+    return float(average_rating)
+
 
+def get_most_watched_genre(user_data):
+    count_watched_genres = {}
+    if user_data["watched"] == []:
+        return None
+    for element in user_data["watched"]:
+        if element["genre"] not in count_watched_genres:
+            count_watched_genres[element["genre"]] = 1
+        count_watched_genres[element["genre"]] += 1
+    sorted_dict = sorted(count_watched_genres.items(), key=lambda item: item[1], reverse=True)
+    return sorted_dict[0][0]
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    unique_set = set()
+    unique_watched = []
+    user_data_watched = user_data["watched"]
+    user_data_friends_list = user_data["friends"]
+    
+    for friend in user_data_friends_list:
+        for i in range(len(friend["watched"])):
+            title = friend["watched"][i]["title"]
+            unique_set.add(title)
+    
+    for my_movie in user_data_watched:
+        if my_movie["title"] not in unique_set:
+            unique_watched.append(my_movie)
         
+    
+    return unique_watched
+
+
+
+def did_user_watch_this_title(user_data, title):
+    for movie in user_data["watched"]:
+        if movie["title"] == title:
+            return True
+    return False
+
+def get_dictionary_of_friends_movies(user_data):
+    dictionary_friends_watched = {}
+    for friend in user_data["friends"]:
+        for i in range(len(friend["watched"])):
+            title = friend["watched"][i]["title"]
+            movie_dict = friend["watched"][i]
+            dictionary_friends_watched[title] = movie_dict
+    
+    return dictionary_friends_watched
+
+def get_friends_unique_watched(user_data):

```

Comment:
Great use of helper functions! It is actually not necessary to put the friends' movies into a dictionary. Instead of comparing strings, you can just check to see if the friend's movie is in the user's watched list:
```
if friend_movie not in user_data["watched"] and friend_movie not in list_of_movies:
    list_of_movies.append(friend_movie)
```